de.metas.acct.base/src/main/sql/postgresql/ddl/functions/AccountSheetReport.sql

```diff
@@ -0,0 +1,235 @@
+                amtacctdr        numeric,
+                amtacctcr        numeric,
+                account_id       numeric,
+                fact_acct_id     numeric,
+                c_doctype_id     numeric,
+                c_tax_id         numeric,
+                c_taxcategory_id numeric,
+                beginningBalance numeric,
+                endingBalance    numeric
+            )
+AS
+$BODY$
+DECLARE
+    v_time                              timestamp;
+    LINE_TYPE_BEGINNINGBALANCE CONSTANT text = 'B';
+    LINE_TYPE_TRANSACTION      CONSTANT text = 'T';
+    v_temp                              numeric;
+BEGIN
+    v_time := logDebug('start');
+
+    --
+    -- create temporary table for everything we're working on; it has no rows, only the needed columns
+    DROP TABLE IF EXISTS tbpFilteredFactAcct;
+    CREATE TEMPORARY TABLE tbpFilteredFactAcct AS
+        (
+            SELECT fa.*,
+                   tc.c_taxcategory_id,
+                   0::numeric beginningBalance,
+                   0::numeric endingBalance,
+                   NULL::text lineType
+            FROM c_elementvalue ev
+                     INNER JOIN fact_acct fa ON ev.c_elementvalue_id = fa.account_id -- only search for accounts with at least 1 transaction
+                     LEFT JOIN c_tax t ON fa.c_tax_id = t.c_tax_id
+                     LEFT JOIN c_taxcategory tc ON t.c_taxcategory_id = tc.c_taxcategory_id
+            WHERE FALSE
+        );
+
+    v_time := logDebug('created empty temporary table', v_time);
+
+
+    --
+    -- insert into the temp table the beginningBalance for all the available accounts which have the balance != 0
+    -- noinspection SqlInsertValues
+    WITH filteredElementValues AS
+             (
+                 SELECT ev.c_elementvalue_id
+                 FROM c_elementvalue ev
+                 WHERE TRUE
+                   AND ev.accounttype = 'A'
+                   AND (p_account_id IS NULL OR ev.c_elementvalue_id = p_account_id)
+                 ORDER BY ev.c_elementvalue_id
+             ),
+         previousBalances AS
+             (
+                 SELECT (de_metas_acct.acctBalanceToDate(ev.c_elementvalue_id, p_c_acctschema_id, (p_dateFrom - INTERVAL '1 day')::date, p_ad_org_id)::de_metas_acct.BalanceAmt).Balance previousDayBalance,
+                        ev.c_elementvalue_id
+                 FROM filteredElementValues ev
+             ),
+         nonZeroPreviousBalances AS
+             (
+                 SELECT *
+                 FROM previousBalances
+                 WHERE previousDayBalance != 0
+             )
+    INSERT
+    INTO tbpFilteredFactAcct (beginningBalance, endingBalance, lineType, account_id)
+    SELECT nonZero.previousDayBalance,
+           nonZero.previousDayBalance,
+           LINE_TYPE_BEGINNINGBALANCE,
+           nonZero.c_elementvalue_id
+    FROM nonZeroPreviousBalances nonZero;
+
+    SELECT count(1) FROM tbpFilteredFactAcct INTO v_temp;
```

Comment:
use 
`GET DIAGNOSTICS v_temp = ROW_COUNT;` instead of actually counting for logging...

Reply:
didn't know about `get diagnostics`.

Reply:
fixed